### ISSUE TITLE
Restore building Python 3.4 wheels

### ISSUE THREAD

**Issue by adamchainz** [scoutapp/scout_apm_python#584]
`cibuildwheel` dropped building Python 3.4 wheels back in November. But we have customers using Python 3.4. We should:

1. Pin the version of `cibuildwheel` we use
2. Add a separate wheel build step in the release flow using Python 3.4 and the last version of `cibuildwheel` that supported it.
3. Make a new release with a Python 3.4 wheel so that customers using Python 3.4 get fast, reliable installs using the extension again.

**Comment by tim-schilling** [comment]
cibuildwheel dropped support for python 3.5 in v2. I suspect python 3.10 will only be available in v2 of cibuildwheel.

**Comment by tim-schilling** [comment]
I spent some time today trying to re-add support for building python 3.4 wheels and was unable to do so. Something downstream seems to have broken: https://github.com/tim-schilling/packaging_tutorial/pull/2. I got stuck with the docker image cibuildwheel would use for cp34-* not being able to find  `bin/pip`.

This can be attempted again in the future by using the following matrix include:

```
          - os: ubuntu-18.04
            python: 3.7
            cibuildwheel_version: "cibuildwheel==0.12.0"
            cibw_build: "cp34-*"
```

It requires the env `CIBW_PLATFORM: ${{ matrix.cibw_platform }}` when running `cibuildwheel`.

This can be used in conjunction with the release.yml file from #681
